api/v1alpha1/map_types.go

```diff
@@ -0,0 +1,214 @@
+	// +optional
+	EvictionPolicy types.EvictionPolicyType `json:"persistenceEnabled,omitempty"`
+
+	// Max size of the map.
+	// +kubebuilder:default:=0
+	// +optional
+	MaxSize *int32 `json:"maxSize,omitempty"`
+
+	// Policy for deciding if the maxSize is reached.
+	// +kubebuilder:default:="PER_NODE"
+	// +optional
+	MaxSizePolicy types.MaxSizePolicyType `json:"maxSizePolicy,omitempty"`
+}
+
+type IndexConfig struct {
+	// Name of the index config.
+	// +optional
+	Name string `json:"name,omitempty"`
+
+	// Type of the index.
+	Type IndexType `json:"type"`
+
+	// Attributes of the index.
+	Attributes []string `json:"attributes"`
+
+	// Options for "BITMAP" index type.
+	// +optional
+	BitmapIndexOptions *BitmapIndexOptionsConfig `json:"bitMapIndexOptions,omitempty"`
+}
+
+// +kubebuilder:validation:Enum=SORTED;HASH;BITMAP
+type IndexType string
+
+const (
+	IndexTypeSorted IndexType = "SORTED"
+	IndexTypeHash   IndexType = "HASH"
+	IndexTypeBitmap IndexType = "BITMAP"
+)
+
+type BitmapIndexOptionsConfig struct {
+	UniqueKey string `json:"uniqueKey"`
+
+	UniqueKeyTransition UniqueKeyTransition `json:"uniqueKeyTransition"`
+}
+
+// +kubebuilder:validation:Enum=OBJECT;LONG;RAW
+type UniqueKeyTransition string
+
+const (
+	UniqueKeyTransitionObject UniqueKeyTransition = "OBJECT"
+	UniqueKeyTransitionLong   UniqueKeyTransition = "LONG"
+	UniqueKeyTransitionRAW    UniqueKeyTransition = "RAW"
+)
+
+// MapStatus defines the observed state of Map
+type MapStatus struct {
+	State          MapConfigState            `json:"state,omitempty"`
+	Message        string                    `json:"message,omitempty"`
+	MemberStatuses map[string]MapConfigState `json:"memberStatuses,omitempty"`
+}
+
+type MapConfigState string
+
+const (
+	MapFailed  MapConfigState = "Failed"
+	MapSuccess MapConfigState = "Success"
+	MapPending MapConfigState = "Pending"
+)
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+
+// Map is the Schema for the maps API
+// +kubebuilder:printcolumn:name="Status",type="string",JSONPath=".status.state",description="Current state of the Map Config"
+type Map struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   MapSpec   `json:"spec"`
+	Status MapStatus `json:"status,omitempty"`
+}
+
+func (m *Map) MapName() string {
+	if m.Spec.Name != "" {
+		return m.Spec.Name
+	}
+	return m.Name
+}
+
+//+kubebuilder:object:root=true
+
+// MapList contains a list of Map
+type MapList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []Map `json:"items"`
+}
+
+func init() {
+	SchemeBuilder.Register(&Map{}, &MapList{})
+}
+
+var EncodeInMemoryFormat map[types.InMemoryFormat]int32 = map[types.InMemoryFormat]int32{
+	types.InMemoryFormatBinary: 0,
+	types.InMemoryFormatObject: 1,
+	types.InMemoryFormatNative: 2,
+}
+
+var EncodeMaxSizePolicy map[types.MaxSizePolicyType]int32 = map[types.MaxSizePolicyType]int32{
```

Comment:
```suggestion
var EncodeMaxSizePolicy = map[types.MaxSizePolicyType]int32{
```
I think this should work. There are other lines in this file that you can apply it.

Reply:
I wasn't aware this worked. I will also put all these fields into a `var` block.